File: utils/permissions.py
import discord
from discord.ext import commands

def has_permissions_to_send_messages():
    async def predicate(ctx):
        if not ctx.channel.permissions_for(ctx.author).send_messages:
            await ctx.author.send("You do not have permission to send messages in this channel.")
            return False
        return True
    return commands.check(predicate)

def has_permissions_to_send_images():
    async def predicate(ctx):
        if not ctx.channel.permissions_for(ctx.author).attach_files:
            await ctx.author.send("You do not have permission to send images in this channel.")
            return False
        return True
    return commands.check(predicate)

def has_permissions_to_use_external_emojis():
    async def predicate(ctx):
        if not ctx.channel.permissions_for(ctx.author).use_external_emojis:
            await ctx.author.send("You do not have permission to use external emojis in this channel.")
            return False
        return True
    return commands.check(predicate)

File: utils/permissions.ts
import { Message } from 'discord.js-selfbot-v13';

export function hasPermissionsToSendMessages() {
    return async function (message: Message) {
        if (!message.channel.permissionsFor(message.author)?.has('SEND_MESSAGES')) {
            await message.author.send("You do not have permission to send messages in this channel.");
            return false;
        }
        return true;
    };
}

export function hasPermissionsToSendImages() {
    return async function (message: Message) {
        if (!message.channel.permissionsFor(message.author)?.has('ATTACH_FILES')) {
            await message.author.send("You do not have permission to send images in this channel.");
            return false;
        }
        return true;
    };
}

export function hasPermissionsToUseExternalEmojis() {
    return async function (message: Message) {
        if (!message.channel.permissionsFor(message.author)?.has('USE_EXTERNAL_EMOJIS')) {
            await message.author.send("You do not have permission to use external emojis in this channel.");
            return false;
        }
        return true;
    };
}
